torch/distributed/algorithms/quantization.py

```diff
@@ -0,0 +1,141 @@
+            f'Quantization type {qtype} is not supported'
+        )
+
+def _quantize_tensor_list(tensor_list, qtype):
+    if not isinstance(tensor_list, list) or not all(
+        isinstance(p, torch.Tensor) for p in tensor_list
+    ):
+        raise RuntimeError(
+            f"_quantize_tensor_list expecting list of torch.Tensor as input but found {type(tensor_list)}"
+        )
+    if (qtype == QuantType.FP16):
+        quantized_tensor_list = [_quantize_tensor(t, qtype) for t in tensor_list]
+        return quantized_tensor_list
+    else:
+        raise RuntimeError(
+            f'Quantization type {qtype} is not supported'
+        )
+
+def _dequantize_tensor(tensor, qtype, quant_loss=None):
+    if not isinstance(tensor, torch.Tensor):
+        raise RuntimeError(
+            f"_dequantize_tensor expecting torch.Tensor as input but found {type(tensor)}"
+        )
+    if (qtype == QuantType.FP16):
+        if tensor.dtype != torch.float16:
+            raise RuntimeError(
+                f"tensor dtype is {tensor.dtype} while expected to be FP16."
+            )
+        elif tensor.dtype == torch.float16 and quant_loss is None:
+            return tensor.float()
+        else:
+            return tensor.float() / quant_loss
+    else:
+        raise RuntimeError(
+            f'Quantization type {qtype} is not supported'
+        )
+
+
+def _dequantize_tensor_list(tensor_list, qtype, quant_loss=None):
+    if not isinstance(tensor_list, list) or not all(
+        isinstance(p, torch.Tensor) for p in tensor_list
+    ):
+        raise RuntimeError(
+            f"_dequantize_tensor_list expecting list of torch.Tensor as input but found {type(tensor_list)}"
+        )
+    elif (qtype == QuantType.FP16):
+        dequantized_tensor_list = [_dequantize_tensor(t, qtype) for t in tensor_list]
+        return dequantized_tensor_list
+    else:
+        raise RuntimeError(
+            f'Quantization type {qtype} is not supported'
+        )
+
+
+def auto_quantize(func, qtype, quant_loss=None):
+    """
+    This is a prototype API that automatically quantize the input tensors, choose the precision types, and
+    pass other necessary arguments and then dequantizes the output.
+
+    Currently it only supports:
+        . FP16 quantization method
+        . all_gather, all_to_all collective ops
+
+    Args:
+        func: (str, name of the collective op) currently supporting dist.all_gather or dist.all_to_all
+        qtype: QuantType.FP16
+
+
+        args[0]: The output which is the tensor_list (list[Tensor]). It should contain
+        correctly-sized tensors to be used for output of the collective.
+        args[1]: The input of the collective.
+            . all_gather: Tensor
+            . all_to_all: List of tensors
+        async_op (bool, optional): currently this is not supported.
+        quant_loss: (float) initialized to None.
+
+    Returns:
+        tensor_list (list[Tensor]): The output of the collective op.
```

Comment:
There's no return "name", just document the return type, and explain what the return type is, the return type for the `auto_quantize` API should be a `Callable` instead. 

Please take a look at an example in c10d API https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L2412